### PR TITLE
S6 sidebar UI

### PR DESCRIPTION
**Pull request by lingsat**

### CODE REVIEW

File: src/utils/products.utils.ts [hunk 1]
```diff
@@ -14,7 +18,81 @@ export const getFilteredProducts = (
...
+        existingCategory.count++;
+      } else {
+        acc.push({ title: product.category, count: 1 });
+      }
+      return acc;
+    },
+    []
+  );
+  return categories;
+};
```
Comment (Yuliia143) [inline src/utils/products.utils.ts]: Can't you use categories list from 'getCategoriesWithCount' ?

File: src/utils/products.utils.ts [hunk 2]
```diff
@@ -14,7 +18,81 @@ export const getFilteredProducts = (
...
+
+export const getBrands = (productsArr: IProduct[]): string[] => {
+  return productsArr.reduce((acc: string[], product) => {
+    if (!acc.includes(product.brand)) {
+      acc.push(product.brand);
+    }
+    return acc;
+  }, []);
+};
```
Comment (Yuliia143) [inline src/utils/products.utils.ts]: Can't you get Brands from 'getCategoriesWithBrands' ?

File: src/features/products/components/ListFilter/ListFilter.tsx [hunk 3]
```diff
@@ -15,12 +36,57 @@ const ListFilter: FC = () => {
...
-        Filter
-      </div>
+        <div className="filter__block">
+          <h3 className="filter__title">Categories</h3>
+          <ul className="filter-categories">
+            {categories.map((category, index) => (
+              <li
+                key={`filterCat-${category}-${index}`}
+                className="filter-categories__item">
+                <button className="filter-categories__btn">
```
Comment (Yuliia143) [inline src/features/products/components/ListFilter/ListFilter.tsx]: Why do you need button here ?

Comment (lingsat) [reply src/features/products/components/ListFilter/ListFilter.tsx]: I don`t - button removed

File: src/utils/products.utils.ts (hunk 1 above)
Comment (lingsat) [reply src/utils/products.utils.ts]: Fixed - create only 1 function that returns { [category]: { count: number, brands: string[] } }. And from that obj i get categories, brands and count

File: src/utils/products.utils.ts (hunk 2 above)
Comment (lingsat) [reply src/utils/products.utils.ts]: Fixed - remove "getCategoriesWithBrands" fn, change "getBrands" fn - it takes as argument categoriesObj and return array of unique brands

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**